lib/collection/src/shards/transfer/driver.rs

```diff
@@ -69,6 +73,18 @@ pub async fn transfer_shard(
             .await?;
         }
 
+        // Transfer shard record in batches for resharding
+        ShardTransferMethod::ReshardingStreamRecords => {
+            transfer_resharding_stream_records(
+                shard_holder.clone(),
+                progress,
+                shard_id,
+                remote_shard,
+                collection_name,
+            )
+            .await?;
+        }
```

Comment:
Even thought it is checked in `validate_transfer`, I'd still make missing `to_shard_id` a hard error for `ReshardingStreamRecords` transfer:

```suggestion
        // Transfer shard record in batches for resharding
        ShardTransferMethod::ReshardingStreamRecords => {
            transfer_resharding_stream_records(
                shard_holder.clone(),
                progress,
                transfer_config.to_shard_id.ok_or_else(|| CollectionError::service_error("to_shard_id is None")),
                remote_shard,
                collection_name,
            )
            .await?;
        }
```

Reply:
Disregard this. There's a further comment that obsoletes this one. Forgot to remove this one before posting a review.